src/factories/meta/SubDivideFunctors.h

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023, Simon Gardner
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+#pragma once
+
+#include <JANA/JApplication.h>
```

Comment:
This file should live in src/algorithms, not in jana land. It can use the algorithms service proxies.

Reply:
An example is in https://github.com/eic/EICrecon/blob/main/src/algorithms/calorimetry/CalorimeterHitDigi.h#L62 and https://github.com/eic/EICrecon/blob/main/src/factories/calorimetry/CalorimeterHitDigi_factory.h#L35.

Reply:
I originally tried this but I'm not sure this how to make this available at the point the lambda/functor is being created

Reply:
Coming back to this a few days later hasn't helped, this is probably a lack of the necessary C++ understanding from me. I don't quite understand how giving the factory:
`Service<AlgorithmsInit_service> m_algorithmsInit {this};`

Allows an algorithm access to:
`const algorithms::GeoSvc& m_geo = algorithms::GeoSvc::instance();`

The way the function is passed as a configuration to the factory generator means that the function has to be made before the factory itself, so `AlgorithmsInit_service` can't yet exist.

Reply:
In short, algorithms are created when the init method of the factory is called, so when the AlgorithmsInit_service has already been instantiated.

Reply:
Ok, moving the initialisation of the variables in `GeometrySplit` which rely on the geometry to the first call of the function rather than it's constructor works fine.

Reply:
Yeah, that's perfect. Only comment I left is to use `call_once` for the flag, which is essentially the same but has more thread safety guarantees.